simphony/io/file_lattice.py

```diff
@@ -0,0 +1,157 @@
+
+    """
+    def __init__(self, file, name, type, base_vect, size, origin,
+                 record=None):
+        self._type = type
+        self._base_vect = np.array(base_vect, dtype=np.float)
+        self._size = tuple(size)
+        self._origin = np.array(origin, dtype=np.float)
+        self._file = file
+        self._group = file.root.lattice
+        self._record = record
+
+        # If Lattice not in file, create a lattice
+        if name not in self._file.root.lattice:
+            if self._record is None:
+                # If record not specified use NoUIDRecord
+                self._record = NoUIDRecord
+            self._table = IndexedDataContainerTable(
+                self._group, name, self._record, np.prod(self._size))
+            for i in xrange(np.prod(self._size)):
+                self._table.append(DataContainer())
+
+            # Set table attributes
+            lattice = self._group._f_getChild(self.name)
+            lattice.attrs.type = self._type
+            lattice.attrs.base_vect = self._base_vect
+            lattice.attrs.size = self._size
+            lattice.attrs.origin = self._origin
+
+        # If Lattice already in file, return a reference
+        else:
+            self._table = IndexedDataContainerTable(self._group, name)
+            lattice = self._group._f_getChild(self.name)
+            self._type = lattice.attrs.type
+            self._base_vect = lattice.attrs.base_vect
+            self._size = lattice.attrs.size
+            self._origin = lattice.attrs.origin
+
+    def get_node(self, index):
+        """Get a copy of the node corresponding to the given index.
+
+        Parameters:
+        -----------
+        index : tuple of D x int (node index coordinate)
+
+        Returns:
+        -----------
+        A reference to a LatticeNode object
+
+        """
+        n = np.ravel_multi_index(index, self._size)
+        return LatticeNode(index, self._table[n])
+
+    def update_node(self, node):
+        """ Updates FileLattice data for a LatticeNode
+
+        Parameters:
+            node : LatticeNode
+                reference to LatticeNode object
+
+        """
+        # Find correct row for node
+        n = np.ravel_multi_index(node.index, self._size)
+        self._table[n] = node.data
+
+    def iter_nodes(self, indices=None):
+        """Get an iterator over the LatticeNodes described by the ids.
+
+        Parameters:
+        -----------
+        indices : iterable set of D x int, optional
+            node index coordinates
+
+        Returns:
+        -----------
+        A generator for LatticeNode objects
+
+        """
+        if indices is None:
+            for row_number, data in enumerate(self._table):
+                index = np.unravel_index(row_number, self._size)
+                yield LatticeNode(index, data)
+        else:
+            for index in indices:
+                yield self.get_node(index)
+
+    def get_coordinate(self, index):
```

Comment:
Please make sure you add some tests for the `OrthotombicP` and `Hexagonal` lattice